### ISSUE TITLE
Corruption of source files using "rsync" to sdfs volume

### ISSUE THREAD

**Issue by tony-travis** [opendedup/sdfs#19]
I've just tried:

`rsync -axP / /mnt`

Where "/mnt" is an sdfs volume and discovered that it systematically corrupts the source files, replacing them with files 61 byes long reported as "Java serialization data, version5". This confirms a bug reported last February on the "sdfs" Google group:

[rsync sdfs BUG report](https://groups.google.com/forum/#!msg/dedupfilesystem-sdfs-user-discuss/ldngtM5psag/v8gduWIjDAAJ)

Yes, I backed up my system before trying "sdfs", but this is still a show-stopper for me :-(

**Comment by opendedup** [comment]
Thanks tony - I will test here.

**Comment by tony-travis** [comment]
Hi, what did you find out?

**Comment by tony-travis** [comment]
Has anyone else confirmed this bug?

**Comment by opendedup** [comment]
Tony - I am able to download the file fine. Are you still having an issue

**Comment by stroucki** [comment]
Noticed it at first when /bin/less was trashed.
Files are corrupted like this:
```
root@debian:/tmp/back# ls -l /usr/share/sdfs/sdfscli
-rwxrwxr-x 1 root root 61 Jul 15 23:32 /usr/share/sdfs/sdfscli
root@debian:/tmp/back# hexdump -C /usr/share/sdfs/sdfscli
00000000  ac ed 00 05 73 72 00 27  6f 72 67 2e 6f 70 65 6e  |....sr.'org.open|
00000010  64 65 64 75 70 2e 73 64  66 73 2e 69 6f 2e 4d 65  |dedup.sdfs.io.Me|
00000020  74 61 44 61 74 61 44 65  64 75 70 46 69 6c 65 c0  |taDataDedupFile.|
00000030  2d 48 41 c2 73 c0 35 0c  00 00 78 70 78           |-HA.s.5...xpx|
0000003d
```
My filesystems:
```
/dev/vdb                                 103081248 1245772  96576212   2% /opt/sdfs/volumes/pool0
sdfs:/etc/sdfs/pool0-volume-cfg.xml:6442  94435840  521728  93914112   1% /tmp/expose
```
root is bindmounted on /tmp/back
rsync command was rsync -aH /tmp/back/ /tmp/expose/1/ &

**Comment by opendedup** [comment]
This issue has been fixed in 3.5.4
